e2e/nomostest/clusters/gke.go

```diff
@@ -58,6 +60,70 @@ func withKubeConfig(cmd *exec.Cmd, kubeconfig string) *exec.Cmd {
 	return cmd
 }
 
+// listOperations lists RUNNING operations on the target cluster
+func listOperations(t testing.NTB, name string) ([]string, error) {
+	args := []string{
+		"container", "operations", "list",
+		"--project", *e2e.GCPProject,
+		"--filter", fmt.Sprintf("status = RUNNING AND targetLink ~ %s", name),
+		"--format", "value(name)",
+	}
+	if *e2e.GCPZone != "" {
+		args = append(args, "--zone", *e2e.GCPZone)
+	}
+	if *e2e.GCPRegion != "" {
+		args = append(args, "--region", *e2e.GCPRegion)
+	}
+	t.Logf("gcloud %s", strings.Join(args, " "))
+	cmd := exec.Command("gcloud", args...)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, errors.Errorf("failed to list operations for %s: %v\nstdout/stderr:\n%s",
+			name, err, string(out))
+	}
+	operations := strings.Fields(string(out))
+	return operations, nil
+}
+
+// waitOperation waits for the provided operation to complete
+func waitOperation(t testing.NTB, operation string) error {
+	args := []string{
+		"container", "operations", "wait",
+		operation,
+		"--project", *e2e.GCPProject,
+	}
+	if *e2e.GCPZone != "" {
+		args = append(args, "--zone", *e2e.GCPZone)
+	}
+	if *e2e.GCPRegion != "" {
+		args = append(args, "--region", *e2e.GCPRegion)
+	}
+	t.Logf("gcloud %s", strings.Join(args, " "))
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
suggest pulling ctx out as a func var so you can define the timeout or cancel externally

Reply:
will do

Reply:
done